BrainPortal/app/controllers/data_providers_controller.rb

```diff
@@ -103,16 +101,16 @@ def create #:nodoc:
       @provider.addlog_context(self,"Created by #{current_user.login}")
       flash[:notice] = "Provider successfully created."
       respond_to do |format|
-        format.js   { redirect_to :action => :index, :format => :js  }
+        format.html { redirect_to :action => :index, :format => :html}
         format.xml  { render :xml   => @provider }
         format.json { render :json  => @provider }
       end
     else
       @typelist = get_type_list
       respond_to do |format|
-        format.js   { render :partial  => "shared/failed_create", :locals => {:model_name => "data_provider"} }
```

Comment:
Can you double-check something? The partial "shared/failed_create" might not bee needed anymore at all.. if it's not referenced anywhere we can delete the file.